### PR TITLE
EZP-31088: Refactored Language GW to use Doctrine\DBAL

### PR DESCRIPTION
**Pull request by alongosz**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31088](https://jira.ez.no/browse/EZP-31088)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `master (8.0@dev)` for eZ Platform `v3.0`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

This PR refactors Content Language gateway to rely on `\Doctrine\DBAL\Connection` instead of our own deprecated `DatabaseHandler`.

Affected implementation:
- `\eZ\Publish\Core\Persistence\Legacy\Content\Language\Gateway`
- `\eZ\Publish\Core\Persistence\Legacy\Content\Language\Gateway\DoctrineDatabase`
- `\eZ\Publish\Core\Persistence\Legacy\Content\Language\Gateway\ExceptionConversion`

Additionally strict types were introduced to the all Language gateway methods and forced via `declare(strict_types=1)` as it seems in this case it doesn't cause issues.

The Language gateway implementations were marked as `@internal` and `final`. The abstract Gateway class has been marked as `@internal`.

What could stand out here is a major refactoring done to the 
`\eZ\Publish\Core\Persistence\Legacy\Content\Language\Gateway\DoctrineDatabase::canDeleteLanguage` 
method. The queries build there follow a very specific pattern, so the code was unified. From the architecture point of view it should be delegated (e.g. dispatched) to other gateways which are responsible for handling specific parts of the system. However for now it's out of scope and would conflict with the other changes to those gateways which need to be done first.

**TODO**:
- [x] Refactor Content Language gateway to use Doctrine\DBAL.
- [x] Replace DatabaseHandler GW constructor argument with the Connection.
- [x] Introduce strict types to Content Language gateway methods.
- [x] Force strict types for Content Language gateway classes.
- [x] Mark Content Language gateway classes as `final` and `@internal`.
- [x] Align tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/API/Repository/Tests/LanguageServiceTest.php [hunk 1]
```diff
@@ -514,6 +511,9 @@ public function testDeleteLanguage()
         // +1 -1
         $this->assertEquals($beforeCount, count($languageService->loadLanguages()));
 
+        $this->expectException(NotFoundException::class);
+        $this->expectExceptionMessage('Could not find \'Language\' with identifier \'eng-NZ\'');
```
Comment (adamwojs) [inline eZ/Publish/API/Repository/Tests/LanguageServiceTest.php]:
```suggestion
        $this->expectExceptionMessage("Could not find 'Language' with identifier 'eng-NZ'");
```

File: eZ/Publish/Core/Persistence/Legacy/Content/Language/Gateway/DoctrineDatabase.php [hunk 2]
```diff
@@ -162,327 +138,91 @@ public function loadLanguageListDataByLanguageCode(array $languageCodes): iterab
...
-     * Loads the data for all languages.
-     *
-     * @return string[][]
-     */
-    public function loadAllLanguagesData()
+    public function loadAllLanguagesData(): array
     {
         $query = $this->createFindQuery();
 
         return $query->execute()->fetchAll();
```
Comment (mikadamczyk) [inline eZ/Publish/Core/Persistence/Legacy/Content/Language/Gateway/DoctrineDatabase.php]: nitpick
```suggestion
         return $this->createFindQuery()->execute()->fetchAll();
```

Comment (alongosz) [reply eZ/Publish/Core/Persistence/Legacy/Content/Language/Gateway/DoctrineDatabase.php]: Fixed via 39a874c.

**Comment by alongosz** [comment]
Note: let's wait with QA for other PRs for the same sub-task

**Comment by micszo** [comment]
Sanities OK.

**Pull request merged**